docs/resources/x509_certificate.md

```diff
@@ -0,0 +1,141 @@
+certificate signing process. It describes which authority is guaranteeing the
+identity of our certificate.
+
+e.g. `/C=US/L=Seattle/CN=Acme Trust CA/emailAddress=support@acmetrust.org`
+
+    describe x509_certificate('/etc/pki/www.mywebsite.com.pem') do
+      its('issuer_cn') { should match "CN=Acme Trust CA" }
+    end
+
+### parsed.XX
+
+`issuer` makes it easier to access individual issuer elements.
+
+    describe x509_certificate('/etc/pki/www.mywebsite.com.pem') do
+      its('issuer.CN') { should eq "Acme Trust CA" }
+    end
+
+### public_key (String)
+
+The `public_key` property returns a base64 encoded public key in PEM format.
+
+    describe x509_certificate('/etc/pki/www.mywebsite.com.pem') do
+      its('public_key') { should match "-----BEGIN PUBLIC KEY-----\nblah blah blah..." }
+    end
+
+### key_length (Integer)
+
+The `key_length` property calculates the number of bits in the public key.
+More bits increase security, but at the cost of speed and in extreme cases, compatibility.
+
+    describe x509_certificate('mycert.pem') do
+      its('key_length') { should be 2048 }
+    end
+
+### signature_algorithm (String)
+
+The `signature_algorithm` property describes which hash function was used by the CA to
+sign the certificate.
+
+    describe x509_certificate('/etc/pki/www.mywebsite.com.pem') do
+      its('signature_algorithm') { should be 'sha256WithRSAEncryption' }
+    end
+
+
+### validity_in_days (Float)
+
+The `validity_in_days` property can be used to check that certificates are not in
+danger of expiring soon.
+
+    describe x509_certificate('/etc/pki/www.mywebsite.com.pem') do
+      its('validity_in_days') { should be > 30 }
+    end
+
+### not_before and not_after (Time)
+
+The `not_before` and `not_after` properties expose the start and end dates of certificate
+validity. They are exposed as ruby Time class so that date arithmetic can be easily performed.
+
+    describe x509_certificate('/etc/pki/www.mywebsite.com.pem') do
+      its('not_before') { should be <= Time.utc.now }
+      its('not_after')  { should be >= Time.utc.now }
+    end
+
+### serial (Integer)
+
+The `serial` property exposes the serial number of the certificate. The serial number is set by the CA during the signing process and should be unique within that CA.
+
+### version (Integer)
+
+The `version` property exposes the certificate version.
+
+### extensions (Hash)
+
+The `extensions` hash property is mainly used to determine what the certificate can be used for.
+
+    describe x509_certificate('/etc/pki/www.mywebsite.com.pem') do
+      its('extensions').length) {should eq 3 }
+
+      # Check what extension categories we have
+      its('extensions) { should include 'keyUsage' }
+      its('extensions) { should include 'extendedKeyUsage' }
+      its('extensions) { should include 'subjectAltName' }
+
+      # Check examples of basic 'keyUsage'
+      its('extensions['keyUsage']) { should include "Digital Signature" }
```

Comment:
Any particular reason you're using hash notation instead of method/dot notation for these examples?